spicy/runtime/src/driver.cc

```diff
@@ -171,12 +172,16 @@ Result<spicy::rt::ParsedUnit> Driver::processInput(const spicy::rt::Parser& pars
 
         in.read(buffer, static_cast<std::streamsize>(len));
 
+        auto profiler = hilti::rt::profiler::start(fmt("spicy/prepare/input/%s", parser.name));
+
         if ( auto n = in.gcount() )
             data->append(hilti::rt::Bytes(buffer, n));
 
         if ( in.peek() == EOF )
             data->freeze();
 
+        hilti::rt::profiler::stop(profiler);
+
```

Comment:
As a style nit, the scope of `profiler` would be clearer easier to understand for me if it relied purely on RAII to trigger `stop`, e.g.,

```suggestion
                {
            auto profiler = hilti::rt::profiler::start(fmt("spicy/prepare/input/%s", parser.name));

            if ( auto n = in.gcount() )
                data->append(hilti::rt::Bytes(buffer, n));

            if ( in.peek() == EOF )
                data->freeze();
        }
```

This wouldn't work everywhere right away since we might need to create variables from inside the scope.

Reply:
The `stop()` is there so that one can insert the instrumentation into existing code with little modification. In this case, the RAII version works pretty easily, too. I'll change.